dora/core/common/src/main/java/alluxio/node/ModularHashingNodeProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.node;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The modular hashing node provider implementation.
+ * @param <T> the type of node
+ */
+public class ModularHashingNodeProvider<T> implements NodeProvider<T> {
+  private List<T> mLastNodes;
+
+  /**
+   * Constructs a new {@link ModularHashingNodeProvider}.
+   *
+   * @param nodes the nodes to select
+   */
+  public ModularHashingNodeProvider(List<T> nodes) {
```

Comment:
what is the use case for this hash+mode algorithm?

Reply:
Modular hash is simple fast and do not care the node id, ConsistentHashing is complex and have to maintain the node list dynamically.

We use these two algorithm to supply a fake file block location, so that the application can dispatch the split to the node what we want.

Reply:
Modular hash does not support dynamic worker membership, so I imagine you have:
1. If you don't expect dynamic worker membership, modular hash or this jump consistent hash https://github.com/Alluxio/alluxio/pull/17817 may be your choice
2. If the worker memberships may change and you want consistent cache hits, consistent hash will be your choice.

I don't think I fully understood how you fake block locations by modular operation. But I think sure you can implement this, and let us know how that works out in your environment.

In https://github.com/Alluxio/alluxio/pull/17817 we have extracted WorkerLocationPolicy interface. Could you move your implementation under `WorkerLocationPolicy `? Thanks!